master/buildbot/worker/protocols/pb.py

```diff
@@ -142,6 +142,7 @@ def loseConnection(self):
             # however, these hacks are pretty internal, so don't blow up if
             # they fail or are unavailable
             log.msg("failed to accelerate the shutdown process")
+        self.notifyDisconnected()
```

Comment:
This seems to be causing test breakage.

Reply:
removed it; needs to be looked at by somebody that is more familiar with the logic